catalogue/webapp/pages/works.js

```diff
@@ -427,8 +435,16 @@ const Works = ({ works }: Props) => {
 
 WorksSearchProvider.getInitialProps = async (ctx: Context): Promise<Props> => {
   const query = ctx.query.query;
+  const dateFrom = makeYearMonthDayString(ctx.query.dateFrom);
+  const dateTo = makeYearMonthDayString(ctx.query.dateTo);
   const page = ctx.query.page ? parseInt(ctx.query.page, 10) : 1;
 
+  function makeYearMonthDayString(year) {
```

Comment:
Part of what we have been doing to is to just pass values from the client to the API, creating this validation here will make things go out of sync pretty quickly.

If the desire to add the year only to the URL is what is wanted, we can do that on the API side, and should probably do it with the parsers we have there to be able to extract dates.

Reply:
I believe we agreed (but will double-check) that the highest resolution required for date search is individual years, so I think maybe only adding the year to the URL is what's required.

Reply:
100% - but we should be making that change on the API side, which I would prefer to do after testing (the point of this exercise is to help us define the API).

But keeping URLs consistent seems cleaner as we don't introduce new ways of doing things.

If it turns out people want to change the URL to just years, we can do that API side and keep the parity.